java/basic_app/app/src/main/java/com/appsflyer/onelink/appsflyeronelinkbasicapp/FruitActivity.java

```diff
@@ -0,0 +1,71 @@
+package com.appsflyer.onelink.appsflyeronelinkbasicapp;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.text.method.ScrollingMovementMethod;
+import android.util.Log;
+import android.view.View;
+import android.widget.TextView;
+import androidx.appcompat.app.AppCompatActivity;
+import com.appsflyer.deeplink.DeepLink;
+import com.google.gson.Gson;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import static com.appsflyer.onelink.appsflyeronelinkbasicapp.AppsflyerBasicApp.LOG_TAG;
+
+public abstract class FruitActivity extends AppCompatActivity {
+    TextView dlAttrs;
+    TextView dlTitleText;
+
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(getLayoutResourceId());
+    }
+
+    protected abstract int getLayoutResourceId();
+
+    protected void setViewVariables(String fruitName) {
```

Comment:
Refactor this method to more generic - use the fruit names dynamically and not with `if`-`else`